src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildListenerTest.java

```diff
@@ -73,6 +81,27 @@ public void onStarted_doesnt_set_if_there_is_no_StashCause() throws Exception {
     verify(build, never()).setDescription(anyString());
   }
 
+  @Test
+  public void onCompleted_checks_build_result() throws Exception {
+    StashBuildTrigger trigger = mock(StashBuildTrigger.class);
+    StashPullRequestsBuilder builder = mock(StashPullRequestsBuilder.class);
+    StashRepository repository = mock(StashRepository.class);
+    FreeStyleProject project = spy(jenkinsRule.createFreeStyleProject());
+    Map<TriggerDescriptor, Trigger<?>> triggerMap = new HashMap<TriggerDescriptor, Trigger<?>>();
+    triggerMap.put(StashBuildTrigger.descriptor, trigger);
+
+    when(build.getCause(eq(StashCause.class))).thenReturn(stashCause);
+    when(build.getParent()).thenReturn(project);
+    when(project.getTriggers()).thenReturn(triggerMap);
+    when(trigger.getBuilder()).thenReturn(builder);
+    when(builder.getRepository()).thenReturn(repository);
+    when(build.getResult()).thenReturn(Result.SUCCESS);
+
+    stashBuildListener.onCompleted(build, taskListener);
+
+    verify(build, times(1)).getResult();
```

Comment:
I think it would be much better to verify that we called `postFinishedComment`.
Just knowing that `getResult` was called doesn't prove much. Posting comments is a user-visible effect.

Reply:
OK

Reply:
Done.